Doc/library/sqlite3.rst

```diff
@@ -47,85 +47,170 @@ This document includes four main sections:
 
-   >>> data = [
-   ...    ('2006-03-28', 'BUY', 'IBM', 1000, 45.0),
-   ...    ('2006-04-05', 'BUY', 'MSFT', 1000, 72.0),
-   ...    ('2006-04-06', 'SELL', 'IBM', 500, 53.0),
-   ... ]
-   >>> cur.executemany('INSERT INTO stocks VALUES(?, ?, ?, ?, ?)', data)
-
-Notice that we used ``?`` placeholders to bind *data* to the query.
+Notice that ``?`` placeholders are used to bind ``data`` to the query.
 Always use placeholders instead of :ref:`string formatting <tut-formatting>`
 to bind Python values to SQL statements,
-to avoid `SQL injection attacks`_.
-See the :ref:`placeholders how-to <sqlite3-placeholders>` for more details.
+to avoid `SQL injection attacks`_
+(see :ref:`sqlite3-placeholders` for more details).
 
-Then, retrieve the data by iterating over the result of a ``SELECT`` statement::
+You can verify that the new rows were inserted
+by executing a ``SELECT`` query,
+this time iterating over the results of the query::
 
-   >>> for row in cur.execute('SELECT * FROM stocks ORDER BY price'):
+   >>> for row in cur.execute("SELECT year, title FROM movie ORDER BY year"):
    ...     print(row)
+   (1971, "And Now for Something Completely Different")
+   (1975, "Monty Python and the Holy Grail")
+   (1979, "Monty Python's Life of Brian")
+   (1982, "Monty Python Live at the Hollywood Bowl")
+   (1983, "Monty Python's The Meaning of Life")
+
+Each row is now a two-item :class:`tuple` of ``(year, title)``.
+
+At last, you'll verify that the database has been written to disk by
+calling :meth:`con.close() <Connection.close>`
+to close the existing connection,
+opening a new one, creating a new cursor,
+then reusing the query from above to read from the database::
+
+   >>> con.close()
+   >>> con2 = sqlite3.connect("tutorial.db")
+   >>> cur2 = con2.cursor()
+   >>> for row in cur.execute("SELECT year, title FROM movie ORDER BY year"):
+   ...     print(row)
+   (1971, "And Now for Something Completely Different")
+   (1975, "Monty Python and the Holy Grail")
+   (1979, "Monty Python's Life of Brian")
+   (1982, "Monty Python Live at the Hollywood Bowl")
+   (1983, "Monty Python's The Meaning of Life")
 
-   ('2006-01-05', 'BUY', 'RHAT', 100, 35.14)
-   ('2006-03-28', 'BUY', 'IBM', 1000, 45.0)
-   ('2006-04-06', 'SELL', 'IBM', 500, 53.0)
-   ('2006-04-05', 'BUY', 'MSFT', 1000, 72.0)
-
-You've now created an SQLite database using the :mod:`!sqlite3` module.
+You've now created an SQLite database using the :mod:`!sqlite3` module,
+inserted data and ran several SQL queries against it.
 
 .. _SQL injection attacks: https://en.wikipedia.org/wiki/SQL_injection
+.. _cursors: https://en.wikipedia.org/wiki/Cursor_(databases)
+.. _flexible typing: https://www.sqlite.org/flextypegood.html
+.. _transactions: https://en.wikipedia.org/wiki/Database_transaction
+.. _sqlite_master: https://www.sqlite.org/schematab.html
+
+.. seealso::
 
+   * :ref:`sqlite3-howtos` for details how to handle specific tasks.
```

Comment:
Do you think it would be helpful to list a few common/useful howtos under this?
We could also include the aforementioned ones about transactions and placeholders, so that users don't have to go back looking for them.

Reply:
It never hurts to link to how-to guides (or indeed anything) from a tutorial.